extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiConfig.java

```diff
@@ -6,104 +6,92 @@
 import java.util.OptionalInt;
 
 import io.quarkus.runtime.annotations.ConfigDocMapKey;
-import io.quarkus.runtime.annotations.ConfigItem;
-import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
 import io.smallrye.openapi.ui.DocExpansion;
 import io.smallrye.openapi.ui.HttpMethod;
 import io.smallrye.openapi.ui.ThemeHref;
 
-@ConfigRoot
-public class SwaggerUiConfig {
+@ConfigMapping(prefix = "quarkus.swagger-ui")
```

Comment:
When there is no `@ConfigRoot`, Quarkus automatically adds one?

Reply:
AFAIU it's only needed when you want to configure it to a different phase than `BUILD_TIME`. 

Right @gsmet ?

Reply:
@mcruzdev you're right, the annotation is still needed. I wrongly assumed it was no longer necessary. Fixed now